ui/src/components/canvas/FlowCanvas.tsx

```diff
@@ -120,11 +135,58 @@ const FlowCanvas = () => {
           onPaneClick={() => clearSelectedChildNode()}
           fitView
         >
-          <Controls>
+          <Controls style={{ bottom: "50px" }}>
+            <ControlButton
+              title="horizontal layout"
+              onClick={() =>
+                updateLayoutOrientation(
+                  "horizontal",
+                  getGraphConstraintOrientation()
+                )
+              }
+            >
+              <ArrowsHorizontal />
+            </ControlButton>
+            <ControlButton
+              title="vertical layout"
+              onClick={() =>
+                updateLayoutOrientation(
+                  "vertical",
+                  getGraphConstraintOrientation()
+                )
+              }
+            >
+              <ArrowsVertical />
+            </ControlButton>
+            <ControlButton
+              title="narrow"
+              onClick={() =>
+                updateLayoutOrientation(getLayoutOrientation(), "narrow")
```

Comment:
The layout orientation already exists in the store, so no need to pass it in here. You can create a new update method in the store to just set the density.